src/cryptography/utils.py

```diff
@@ -51,13 +51,19 @@ def int_from_bytes(data, byteorder, signed=False):
         return int(bytes(data).encode('hex'), 16)
 
 
-def int_to_bytes(integer, length=None):
-    hex_string = '%x' % integer
-    if length is None:
-        n = len(hex_string)
-    else:
-        n = length * 2
-    return binascii.unhexlify(hex_string.zfill(n + (n & 1)))
+if hasattr(int, "to_bytes"):
+    def int_to_bytes(integer, length=None):
+        return integer.to_bytes(
+            length or (integer.bit_length() + 7) // 8 or 1, 'big'
```

Comment:
I'm not positive the `or 1` is correct; I see that it matches the old behavior, but we have exactly one caller that doesn't pass an explicit length, and I think it'd be ok with `b""` as the result here.

@reaperhulk opinions?